nexus/src/app/saga.rs

```diff
@@ -79,12 +77,28 @@ use uuid::Uuid;
 pub(crate) fn create_saga_dag<N: NexusSaga>(
     params: N::Params,
 ) -> Result<SagaDag, Error> {
-    let builder = DagBuilder::new(SagaName::new(N::NAME));
-    let dag = N::make_saga_dag(&params, builder)?;
-    let params = serde_json::to_value(&params).map_err(|e| {
-        SagaInitError::SerializeError(String::from("saga params"), e)
-    })?;
-    Ok(SagaDag::new(dag, params))
+    N::prepare(&params)
```

Comment:
For some reason, the one free function here irked me and I put the impl into a default impl on `NexusSaga`.  This is basically an unrelated change, though.  If we like this, I can change the other callers of `create_saga_dag()`, too.  If we don't, I can give this function a better name and just use it directly from the background tasks that are using `prepare()`.